src/services/textChanges.ts

```diff
@@ -483,107 +483,150 @@ namespace ts.textChanges {
+                //  |-separator
+                startPos = nextToken.end;
+                prefix = spaces(lineAndCharOfNextElement.character - lineAndCharOfNextToken.character);
+            }
+            else {
+                // next element is located on different line that separator
+                // let insert position be the beginning of the line that contains next element
+                startPos = getStartPositionOfLine(lineAndCharOfNextElement.line, sourceFile);
+            }
+
+            // write separator and leading trivia of the next element as suffix
+            const suffix = `${tokenToString(nextToken.kind)}${sourceFile.text.substring(nextToken.end, containingList[index + 1].getStart(sourceFile))}`;
+            this.replaceRange(sourceFile, createTextRange(startPos, containingList[index + 1].getStart(sourceFile)), newNode, { prefix, suffix });
+        }
+
+        private insertNodeInListAfterLast(sourceFile: SourceFile, containingList: NodeArray<Node>, after: Node, index: number, newNode: Node): void {
+            const end = after.getEnd();
+            const afterStart = after.getStart(sourceFile);
+            const afterStartLinePosition = getLineStartPositionForPosition(afterStart, sourceFile);
+
+            const { multilineList, separator } = this.getMultilineAndSeparatorOfList(sourceFile, containingList, afterStartLinePosition, index - 1, after);
+
+            if (multilineList) {
+                // insert separator immediately following the 'after' node to preserve comments in trailing trivia
+                this.replaceRange(sourceFile, createTextRange(end), createToken(separator));
+                // use the same indentation as 'after' item
+                const indentation = formatting.SmartIndenter.findFirstNonWhitespaceColumn(afterStartLinePosition, afterStart, sourceFile, this.formatContext.options);
+                // insert element before the line break on the line that contains 'after' element
+                let insertPos = skipTrivia(sourceFile.text, end, /*stopAfterLineBreak*/ true, /*stopAtComments*/ false);
+                if (insertPos !== end && isLineBreak(sourceFile.text.charCodeAt(insertPos - 1))) {
+                    insertPos--;
                 }
+                this.replaceRange(sourceFile, createTextRange(insertPos), newNode, { indentation, prefix: this.newLineCharacter });
             }
-            return this;
+            else {
+                this.replaceRange(sourceFile, createTextRange(end), newNode, { prefix: `${tokenToString(separator)} ` });
+            }
+        }
+
+        private getMultilineAndSeparatorOfList(sourceFile: SourceFile, containingList: NodeArray<Node>, afterStartLinePosition: number, sampleIndex: number, sampleNode: Node) {
+            let separator: SyntaxKind.CommaToken | SyntaxKind.SemicolonToken;
+            let multilineList = false;
+
+            // insert element after the last element in the list that has more than one item

```

Comment:
I'm not sure this comment makes sense here.  Does it need to be moved or rephrased?